src/core/dev_api/openvino/core/descriptor_tensor.hpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/descriptor/tensor.hpp"
+
+namespace ov {
+namespace descriptor {
+
+// To change Tensor element type please change the Parameter type.
+void set_element_type(Tensor& tensor, const element::Type& elemenet_type);
```

Comment:
should we deprecate set_element_type and set_tensor_type?

Reply:
no, we just hide them from public users